pravega-cli/src/main/java/io/pravega/tools/pravegacli/commands/controller/ControllerCommand.java

```diff
@@ -49,7 +61,29 @@
         ClientConfig clientConfig = new ClientConfig();
         clientConfig.register(JacksonJsonProvider.class);
         clientConfig.property("sun.net.http.allowRestrictedHeaders", "true");
-        Client client = ClientBuilder.newClient(clientConfig);
+
+        Client client;
+
+        // If tls parameters are configured, set them in client
+        if (getCLIControllerConfig().isTlsEnabled()) {
+            KeyStore ks = null;
+            try {
+                ks = KeyStore.getInstance("JKS");
+                ks.load(new FileInputStream(new File(getCLIControllerConfig().getTruststore())), null);
+            } catch (IOException | NoSuchAlgorithmException | CertificateException | KeyStoreException e) {
```

Comment:
Do we want to treat all these exception in the same way? Maybe we could add some hints in the message so the user knows what has gone wrong.

Reply:
Each exception now provides it with its own unique message to give a hint as to why the exception might have been thrown.